app/server/models/credit_transfer.py

```diff
@@ -172,19 +175,77 @@ def resolve_as_completed(self, existing_blockchain_txn=None):
             if self.recipient_user and self.recipient_user.transfer_card:
                 self.recipient_user.transfer_card.update_transfer_card()
 
-        if self.transfer_type == TransferTypeEnum.DEPOSIT and self.fiat_ramp:
+        if self.fiat_ramp and self.transfer_type in [TransferTypeEnum.DEPOSIT, TransferTypeEnum.WITHDRAWAL]:
             self.fiat_ramp.resolve_as_completed()
 
         if not existing_blockchain_txn:
             self.send_blockchain_payload_to_worker()
 
     def resolve_as_rejected(self, message=None):
+        if self.fiat_ramp and self.transfer_type in [TransferTypeEnum.DEPOSIT, TransferTypeEnum.WITHDRAWAL]:
+            self.fiat_ramp.resolve_as_rejected()
+
         self.resolved_date = datetime.datetime.utcnow()
         self.transfer_status = TransferStatusEnum.REJECTED
 
         if message:
             self.resolution_message = message
 
+    def get_transfer_limits(self):
+        relevant_limits = []
+        for limit in LIMITS:
+            applied = limit.application_filter(self)
+            if applied and str(self.transfer_type) in limit.applied_to_transfer_types:
+                relevant_limits.append(limit)
+
+        return relevant_limits
+
+    def check_sender_transfer_limits(self):
+        if self.sender_user is None:
+            # skip if an exchange
```

Comment:
is exchange the only case where `sender_user` would be None? is there any other way to check that it's an exchange? and would we not want to limit exchanges potentially?

Reply:
Maybe just change the comment to 'skip if there is no sender, which implies system send'

> would we not want to limit exchanges potentially?
We might, but you'd put the limit  on the inbound of the two credit transfers

>is there any other way to check that it's an exchange
I think there's an exchange TranferType right @tristanhcole ?

Reply:
There is a TransferType I could use to check instead, but yeah, some sends (such as system or disbursement) only have a transfer account, no user?

Reply:
Comment updated